sw/device/tests/usbdev_vbus_test.c

```diff
@@ -0,0 +1,116 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// USB VBUS test
+//
+// Checks for the presence of VBUS from the host/DPI model, optionally checking
+// that VBUS is low initially when the test starts.
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_usbdev.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/pinmux_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#define USBDEV_BASE_ADDR TOP_EARLGREY_USBDEV_BASE_ADDR
+
+/**
+ * Are we expecting VBUS to be low at the start of the test, ie. no connection
+ * to the physical host or the DPI model is inactive?
+ * Typically useful only for an FPGA target that is being manually supervised.
+ */
+static const bool kCheckLowFirst = false;
+
+/**
+ * USB device handle
+ */
+static dif_usbdev_t usbdev;
+
+/**
+ * Pinmux handle
+ */
+static dif_pinmux_t pinmux;
+
+// Wait with timeout until the VBUS/SENSE signal is in the expected state.
+static status_t vbus_wait(bool expected, uint32_t timeout_micros) {
```

Comment:
I think this function is a copy of the one in the pullup test? Could this factored-out in some usb util library?

Reply:
Since I'm still uncertain of the exact requirements and the environment in which SiVal tests are required to run, I've opted to keep them as simple and self-contained as much as possible for now.
You're right that it is the same function, and there will be a couple of more involved tests created soon, at which point the set of functions required will be clearer and I'd like to move them to a common home then.